servicetalk-concurrent-test-internal/src/main/java/io/servicetalk/concurrent/test/internal/TestPublisherSubscriber.java

```diff
@@ -198,6 +199,7 @@ public T takeOnNext() {
         } finally {
             if (interrupted) {
                 Thread.currentThread().interrupt();
+                throwException(new InterruptedException());
```

Comment:
This method suppresses `InterruptedException` in line 192, but we try to avoid doing that. Consider changing this method to rethrow immediately after catching the first `InterruptedException`.